src/binarylane/config/repository.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+from argparse import Namespace
+from pathlib import Path
+from typing import List, Optional, Type, TypeVar
+from binarylane.pycompat.typing import Protocol
+
+import binarylane.config.sources as src
+from binarylane.config.options import OptionName
+
+
+class Source(Protocol):
+    def get(self, name: str) -> Optional[str]:
+        ...
+
+
+T = TypeVar("T", bound=Source)
+
+
+class Repository:
+    _sources: List[Source]
+
+    def __init__(self, *, default_source: bool = True) -> None:
+        self._sources = []
+
+        if default_source is True:
+            self.add_source(src.DefaultSource())
+
+    def initialize(self, *, commandline: Namespace, config_file: Optional[Path] = src.FileSource.DEFAULT_PATH) -> None:
+        try:
+            # Check for existing commandline to see if we have initialized previously
+            self.get_source(src.CommandlineSource)
+            # If we have, add second command line source
```

Comment:
It isn't clear from the comment why we need to add the commandlinesource if it already exists.